docs/lofi-hifi-backends.md

```diff
@@ -0,0 +1,176 @@
+# Introduce Lofi and Hifi Commands for Key ilab functions to Increase Model Fidelity
+
+
+This document describes adding different data generation, mixing, and training backends to ilab to enable higher fidelity training using the backend code.
+
+
+Currently all training is done via qlora or the like. Adding the following commands will enable higher fidelity training and introduce commands such as data mixing.
+
+
+## Key Component
+
+
+### Building off of the InstructLab Structure Redesign
+
+
+After github.com/instructlab/instructlab/pull/990 is merged, ilab will use a parent -> child command structure. This proposal operates under that new structure.
+
+
+This will be the new structure:
+
+
+```console
+ilab
+|
+|_______model
+|       |
+|       |____convert
+|       |____download
+|       |____train (--convert)
+|       |____serve   (-i)
+|       |____chat
+|       |____inference
+|       |
+|_______data
+|       |
+|       |____generate
+|       |
+|_______config
+|       |
+|       |____init
+|       |
+|_______tax
+|       |
+|       |____diff
+|       |____check
+|       |____download
+```
+
+
+
+
+And this Will be the structure after these new commands are added:
+
+
+
+```console
+ilab
+|
+|_______model
+|       |
+|       |____convert
+|       |____download
+|       |____train (--convert)
+|       |    |
+|       |    |______lofi *
+|       |    |______hifi *
```

Comment:
I think lofi and hifi probably make sense w.r.t. the models themselves. However, currently the size of the model is not the only differentiator between the correlating CLI and backend components. Many of the full-size workflows run a bunch of stuff that the CLI does not currently, and might not ever on a laptop.

Initially, we joked about specifying —turbo to run the backend version of a command, and using the laptop version by default. Whatever it is, I think the name should indicate that it’s not just using a higher fidelity model, but actually just a bigger workload in general.